libs/damap/src/lib/components/dmp/data-storage/storage/storage.component.ts

```diff
@@ -48,8 +48,17 @@ export class StorageComponent implements OnInit {
   }
 
   private getInternalStorages() {
-    if (this.internalStoragesLoaded !== LoadingState.LOADED) {
-      this.store.dispatch(loadInternalStorages());
-    }
+    this.store.dispatch(loadInternalStorages());
+  }
+
+  get activeStorages() {
+    return this.internalStorages.filter(storage => storage.active);
+  }
+
+  public getStorageTitle(storage: InternalStorage) {
+    const translation = storage.translations.find(
+      t => t.languageCode === 'eng',
+    );
+    return translation ? translation.title : storage.translations[0].title;
```

Comment:
major: we should ensure that there is at least one translation before showing a storage option (so we might filter beforehand and never get to this point). Since it is a valid database state to have a storage without translations, this case should also be handled. From the back end standpoint, it will just return the storage with empty translations. The front end might will have to deal with this though. What do you think?

Reply:
I actually do not agree on this point. Right now different validation mechanisms are in place to ensure that at least one translation should always exist (no have a title to be displayed in the frontend). Removing this condition would lead to additional (unnecessary) work, and users could for instance not remove a storage from their DMP anymore.